examples/VB_ASR/infer_constitutive_law_ASR_RCA.py

```diff
@@ -0,0 +1,286 @@
+import sys
+
+if "./examples/VB_ASR/" not in sys.path:
+    sys.path.append("./examples/VB_ASR/")
+
+from helpers import *
+from bayem import vba, MVN, Gamma
+
+eps_0 = 0.0
+X0 = [0.1825, 38.21, 167.9]
+return_jacobian = True
+_path = f"./examples/VB_ASR/results/"
+make_path(_path)
+
+###############################
+# DATA
+###############################
+
+# ts_data = np.array(
+#     [0, 8, 14, 19, 26, 46, 84, 123, 140, 173, 197, 200, 235, 252, 264, 284, 322, 361]
+# )
+# epss_data = [
+#     0,
+#     0.01,
+#     0.01,
+#     0,
+#     0,
+#     0.01,
+#     0.02,
+#     0.01,
+#     0.04,
+#     0.08,
+#     0.09,
+#     0.09,
+#     0.11,
+#     0.13,
+#     0.12,
+#     0.13,
+#     0.14,
+#     0.13,
+# ]
+
+ts_data = np.array(
+    [0, 8, 14, 19, 26, 46, 84, 123, 140, 173, 197, 200, 235, 252, 264, 284, 322, 361]
+)
+epss_data = [
+    0.000,
+    0.000,
+    0.000,
+    0.000,
+    0.000,
+    0.000,
+    0.000,
+    0.000,
+    0.027,
+    0.059,
+    0.092,
+    0.126,
+    0.143,
+    0.155,
+    0.161,
+    0.172,
+    0.182,
+    0.185,
+]
+
+###############################
+# MODEL
+###############################
+
+
+def eps_total(t, eps_inf=X0[0], Tau_c=X0[1], Tau_L=X0[2]):
+    if return_jacobian:
+        z, dz = zeta_func(t, Tau_c, Tau_L)
+        diffs = [z, eps_inf * dz[0], eps_inf * dz[1]]
+        return eps_0 + eps_inf * z, np.array(diffs).T
+    else:
+        z = zeta_func(t, Tau_c, Tau_L)
+        return eps_0 + eps_inf * z
+
+
+def zeta_func(t, Tau_c, Tau_L):
+    a1 = np.exp(-t / Tau_c)
+    a2 = np.exp((-t + Tau_L) / Tau_c)
+    if return_jacobian:
+        d1 = -t * a1 / (Tau_c**2 * (a2 + 1.0))
+        d1 += (1.0 - a1) * (-t + Tau_L) * a2 / (Tau_c**2 * (a2 + 1.0) ** 2)
+        d2 = -(1.0 - a1) * a2 / (Tau_c * (a2 + 1.0) ** 2)
+        return (1.0 - a1) / (1.0 + a2), np.array([d1, d2])
+    else:
+        return (1.0 - a1) / (1.0 + a2)
+
+
+###############################
+# OBJECTIVE (MODEL ERROR)
+###############################
+
+
+def me_vec(X):
+    eps_inf, Tau_c, Tau_L = X
+    if return_jacobian:
+        f, diff = eps_total(ts_data, eps_inf=eps_inf, Tau_c=Tau_c, Tau_L=Tau_L)
+        return np.array(f) - np.array(epss_data), diff
+    else:
+        f = eps_total(ts_data, eps_inf=eps_inf, Tau_c=Tau_c, Tau_L=Tau_L)
+        return np.array(f) - np.array(epss_data)
+
+
+def me_scalar(X):
+    if return_jacobian:
+        f, diff = me_vec(X)
+        return 0.5 * np.dot(f, f), f @ diff
+    else:
+        f = me_vec(X)
+        return 0.5 * np.dot(f, f)
+
+
+###############################
+# VARIATIONAL BAYESIAN (VB)
+###############################
+
+prior_mvn = MVN(mean=X0, precision=np.diag([(abs(x)) ** (-2) for x in X0]))

```

Comment:
How did you get these values for the prior of the parameters and the noise?

Reply:
I did not include this into the script and only put the priors right away. The estimation of priors' parameters was according to some functionalities that I developed myself. I'll check if they are part of Bayem, then I can use/include them as well, otherwise, I think the Bayem should first be modified in that respect (makes sense to facilitate setup of priors in Bayem).

Reply:
I updated it and should be now more clear.

Reply:
I'm not sure I understand the physics behind the choice of the how you determine the prior variance of your parameters?

Reply:
For the model parameters, I did not consider any physics knowledge. Basically I am saying there, the Std. of each parameter is the absolute value of it. So, quite a random choice and relatively non-informative, since such Std. obviously contains samples of an opposite sign (if the parameter is supposed to be positive, the prior even contains negative values, and vice versa). So, it is kind of non-informative I would say. Or ... ?

Reply:
but that allows all the parameters to become negative, which might lack a physical meaning

Reply:
In general yes. My interpretation is from a mathematical perspective though. We have a parametric model and the uncertainty of the parameters is taken to be quite large, covering a lot of samples that potentially are physically-meaningless. IF the model is going to adopt such negative physically-meaningless values for the parameters, AND still is able to represent/simulate the data with quite a good precision, then I would say the model is not very well established (parameterized). Would that make sense?
Another aspect is however that, a prior containing meaningless samples (samples that we are sure they cannot explain the data) will somehow remove the advantage of our expert knowledge, and increase the uncertainty of posteriors for no good reason. This point is nevertheless from a Bayesian point of view. So, at the end it is like being too lazy to come up with as informative a prior as possible.

Reply:
I agree, but in general the idea of Bayesian methods is to come up with priors that restrict the model to meaningful parameters (if we already know that some combinations are not physical, this should be reflected in the choice of the prior). And yes, this increases the posterior accuracy, but that is exactly the point of Bayesian inference.